src/rospkg/os_detect.py

```diff
@@ -126,6 +139,47 @@ def get_codename(self):
             return self.lsb_info[2]
         raise OsNotDetected('called in incorrect OS')
 
+class FdoDetect(OsDetector):
```

Comment:
I'm happy to change it to OsRelease or similar if that makes more sense.  I was trying to follow the LsbDetect multi-platform detector naming scheme, but it really doesn't matter to me what we call it.